### PR TITLE
[AOI]: Remove from mainline

### PR DESCRIPTION
**Pull request by nemaara**
See here: http://forums.wesnoth.org/viewtopic.php?f=2&t=51139.

### CODE REVIEW

**Comment by stevecotton** [comment]
Technical comment: remember the `po/wesnoth-aoi` directory.

How are you collecting opinions/votes about keeping the campaign in mainline? Feedback here, or feedback in the thread?

**Comment by nemaara** [comment]
This actually has a bit of work left, I need to find a new home for the portraits and story image (likely TSG for the story image and elves, and DiD for the silver mage portrait). So when I do that, I just need to remove the po/ folder too?

Feedback written here or in that thread are both fine. 

Just to be clear, doing a vote-based system for making these decisions is a pretty bad idea. If we want to debate about which campaigns should be either removed or rewritten, it should be based on the merit of the campaign itself (and not based on its history of being in mainline either).

**Comment by stevecotton** [comment]
The story image wouldn't fit in TSG - the blade shows a reflection of an orc. This feels like 

There's a corresponding change needed in po/CMakeLists.txt, probably something for scons too.

**Comment by CelticMinstrel** [comment]
Would the story image maybe fit in SotBE? I guess maybe there aren't enough trees there... maybe HTTT?

Also @stevecotton it looks like you forgot to finish writing your post.

**Comment by CelticMinstrel** [comment]
Using the lord portrait for a marshal is interesting... but I suppose it can work here.

**Comment by stevecotton** [comment]
An experienced silver mage, in her own manor, dies instead of teleporting away? That doesn't sound right to me, I'd expect her to turn up with the paladins in the next scenario, and for the AI of the next scenario to know about the importance of the book.

I feel this PR is a case of "throwing stuff over the wall", focusing on doing the removal rather than improving the places where the stuff is landing.

**Comment by CelticMinstrel** [comment]
The changelog is kinda funny, framing it as a move rather than a removal.

**Comment by nemaara** [comment]
> An experienced silver mage, in her own manor, dies instead of teleporting away?

By this logic, one can imagine the extreme case where every silver mage (lore-wise) could just teleport away before being dealt a fatal blow, thus gaining incredible plot armor and being virtually invincible with regard to the storyline. 

You could come up with a plethora of reasons why she didn't teleport away. Maybe she was confident in her skills and two measly necromancers (one of which she obviously has dealt with in the past and knows the limit of his skills) don't seem like a big threat to her. Maybe she didn't think Darken Volk would actually kill her. Maybe teleportation is something that requires a bit of time and preparation, and she didn't have time to accomplish that before they got to her (we certainly avoid explaining its mechanics in mainline). 

Instead of arguing against every change or reassignment of assets by saying you don't feel it's good, why don't you make a suggestion for an alternative? Right now, my impression (even though it may be wrong) is that you're arguing here just for the sake of it rather than _working with me_ to try to improve mainline. Can we try working together? I'm willing to make compromises and listen, as long as there are other ideas floating around.

**Comment by stevecotton** [comment]
At least for the 1.16 branch, would be to do a minimal change of adding some description to the campaign selection dialogue and leaving it to the player to choose whether to skip the campaign.

- (Rookie level, 7 scenarios, minimal storyline so players wanting plot arcs should skip it)

**Comment by nemaara** [comment]
I'm sure you can tell for yourself why that's not a good idea.

Compromising doesn't mean taking the easiest way out out of laziness. It can only happen when everyone involved clearly understands what, fundamentally, we are trying to accomplish and what is fundamentally wrong with our current situation. Clearly, this is not the case here.

**Comment by stevecotton** [comment]
No, I can't see why it's worse than removal 3 weeks before the string-freeze.

You say that you want "compromise", but would you accept any situation that doesn't remove it from mainline?

**Comment by nemaara** [comment]
Yes. If you fundamentally understand why I want to remove it, I'm sure you can come up with a solution.

**Comment by stevecotton** [comment]
You've suggested I'm lazy, you've brought up the phrases "compromise" and "work with me", but then pushed to me the task of finding a way to solve the problems. The discussion threads had a lot of discussion about moving stuff to other places in the new-player area, but for 1.16 we can only look at solutions that will be ready in the next 3 or so weeks.

Adding the silver mage to DiD sounds a good idea, but it should be done as a DiD-focused change focussing on how she improves DiD, rather than as a place to put something from AOI.

**Comment by nemaara** [comment]
Because every solution you propose shows that you don't fundamentally understand why I'm asking for AOI to be removed. In fact, I feel quite disrespected that you continuously try to keep proposing the same things without trying to understand my perspective. On the flip side, I think I have made an effort before to understand yours, and right now I'll try again.

If I'm understanding correctly, you wanted to keep AOI as a simple, novice level campaign that has some gameplay scenarios that rookie players may not have encountered before. This would make the game more friendly to new players, since mainline campaigns can be a bit unfriendly due to their difficulty. I thought that was reasonable, so I initially made the proposal to keep some interesting scenarios from AOI as extra "advanced tutorial" scenarios, or you could even make the proposal to have a separate "Player vs. AI" in the singleplayer menu. 

You could also make the proposal that we remove some of the scenarios in AOI that do not offer said gameplay, or you could even make the proposal that we add some of these to HttT and make Siege of Elensefar easier in order to be friendlier to new players. We could also make some of the novice or rookie campaigns easier and add more tactical guidance for new players. Of course I don't particularly like any of these suggestions (I would rather remove AOI outright), but I would be willing to compromise and go with them. 

Instead, the first suggestion you make is this:

> (Rookie level, 7 scenarios, minimal storyline so players wanting plot arcs should skip it)

And then when I ask you to come up with another solution, you refuse to do so. I wanted _you_ to come up with one of the solutions above to show that you were willing to at least consider things from my perspective, but apparently that was too much to ask for.

If you want to talk about why I chose to add the silver mage to DiD, there are many, many explanations I can get into about how the choice (I in particular) make when designing campaigns, and why that fits better in DiD. One is, I think it's cool to have a bit of extra female representation among mages, so we don't just have the mage class being "a bunch of old, rich men" as a relatively generic representation. Another reason is I'd like to be more careful about what kinds of units appear in said campaigns. The lore for the Great Mage suggests that it (lore-wise) is something extremely unique, so having a Great Mage be there, and so summarily dispensed with by Malin and Darken Volk is less preferable as a design choice (again, for campaigns that I work on). From a gameplay perspective, the damage output of a Great (or even Arch) Mage is annoying to deal with, and not because it's strategically interesting. You basically have to put a unit to bait it out under the pretty likely expectation that the unit will die. Then, you can use your other units to run in and finish the job. This is less of an issue with a Silver Mage, since its ranged attack is only 9-4.

Again, these are all things that I would have hoped anyone reviewing this PR could think about for themselves. For the record, the portrait moves to TSG and DiD are finished in this PR, and having playtested them yesterday, I personally think they fit reasonably well. If you want to suggest a better place to put the portraits, I would also be happy to implement your idea too, so long as it is reasonable.

**Comment by stevecotton** [comment]
You're understanding me correctly, that I want to keep it for rookie players. There's a second point that I haven't seen any response to, which is that it's already been translated into many languages - there's a silver lining to the weakness of having so little plot that it has less translatable strings than some single scenarios of other campaigns.

Adding the silver mage to DiD sounds a good idea. But please let's do it in separate PR focussing on how she improves DiD, rather than as part of the remove-AOI PR. 

> I wanted you to come up with one of the solutions above to show that you were willing to at least consider things from my perspective, but apparently that was too much to ask for.

I don't think you realise the amount of effort that I'm spending on this.

* Adding a new "Player vs AI" button seems the opposite direction to #4531, so I wouldn't have expected you to support it. I don't support it, it seems worse than having an extra campaign in the campaign list.

* The advanced tutorial was discussed in the forum thread, but it's not something that can be done in 3 weeks. I already closed my PRs #4205 and #4399 because I couldn't see them being ready in time.

* For Siege of Elensefar, I think that's the point that many players realise that they needed to play the previous scenarios to win the campaign, not just to win the current battle. Making SoE easier could mean that players continue without picking that up, I think SoE being a barrier is a good thing. I'd suggest putting a "would you have been able to win this battle without the bonus that we've just given you? Well, other campaigns would test your preparations." narrator message in an earlier campaign, and the place that springs to mind for it is AOI S7.

* Removing scenarios from AOI - I wasn't expecting this to be acceptable to you. I'd keep at least S1, S2, S4, S6 and would still need a finale, so that's at least 5 of the 7 scenarios. Removing S5 would need the silver mage intro to be moved to another scenario. Removing anything would need a rebalance, but that can happen during the string-freeze.

> Can we try to work together and find a reasonable compromise (meaning, one that at least considers the core reasons why I've proposed this removal)?

My proposal was to suggest that anyone who wants story and lore should choose to skip this campaign, and we trust the player to make that judgement. As far as I've seen, the only non-story, non-lore reasons are these:

> First, just for fun, we can count the number of "defeat the enemy leader" scenarios in each of these campaigns. Of the 7 in AOI, all 7 have this as the objective. None of the 7 have more than rudimentary tactical diversity (e.g. playing with a few terrains, but no special map design for tactical purposes that I could really see). AOI also uses only standard Wesnoth units, so it does not include extra RPG elements from a unit perspective either. Based on this, I think that AOI does not meet the standard of having "polished gameplay".

This is intended as a rookie campaign, where lining up on the edge of forests and rivers is a recently-introduced tactic. But we've already had that discussion in the thread, and then answer seems to be to create an advanced tutorial, which brings us back to the string-freeze deadline. If there are other core reasons, then please point them out.

**Comment by Pentarctagon** [comment]
Since the string freeze is bring brought up as potentially a significant problem, @Vultraz can you confirm that this is still the goal?  IIRC the reasoning behind that was to be able to ship 1.16 in time for the next Ubuntu LTS release, but if there's a strong enough reason to push that back, it could be done.

Also, to add my 2 cents, I think whether the campaign is fully translated or not is a moot point.  If it's kept, then there's no work to do with translations.  If it's not worth keeping, then whether stuff not worth keeping is translated or not is irrelevant.

**Comment by nemaara** [comment]
Well I still think you're missing the point about why I want to remove AOI, but for now let's try to work out a middle ground solution. A more in depth discussion may be better for somewhere like IRC/Discord. 

I was hoping I was understanding you correctly about why you wanted to keep it. So right now, it looks like (to me) the best compromise we have is making an advanced tutorial. While Pentarctagon had a point on the forum where he said that it might be annoying to new players to have the tutorial be so long, I was thinking if we go along the lines of celmin's idea (splitting the tutorial into basic and advanced based on difficulty), that would be less of a problem. We recommend the basic tutorial to all new first time players, and the advanced tutorial remains there as a choice, much like AOI is a choice to them now. Players can simply access it by select it from the difficulty menu.

Going down the list of scenarios, I think S2 and S4 are the best candidates for adding into an advanced tutorial. Any opinions on that? I already said I was willing to make the scenarios, and it shouldn't take me more than about a week to do so. All I need to do is put in Konrad/Li'sar instead of Erlornas and balance the scenarios properly. If you think those scenarios are the best to put it, I'll get to work on that.

Regarding translations, while we definitely appreciate the translators' work in the past having worked on this campaign, I agree with Pentarctagon. I think we have to be willing to make some sacrifices moving forward (especially if we start considering reworking more campaigns). Furthermore, there aren't really that many translatable strings in AOI compared to something like TSG or DiD (which you note I rewrote, with many hundreds of strings, sorry to annoy translators), so I think the amount of work lost is not very much. 

Regarding the Silver Mage in DiD, I don't mind moving it to another place, but I've been told in the past to make all relevant changes (e.g. moving assets) in one place, like a single commit or a single PR. I thought that that also applied here to moving the portrait to DiD, but if I'm wrong, I'm fine with making a new PR for that.

**Comment by stevecotton** [comment]
If someone else has told you to do PRs in a different way then they're probably right and I'm wrong. I think it would still be worth doing the text changes in a separate PR, using the default silver mage portrait.

**Comment by CelticMinstrel** [comment]
> Adding the silver mage to DiD sounds a good idea. But please let's do it in separate PR focussing on how she improves DiD, rather than as part of the remove-AOI PR.

Personally I'd prefer to do it as part of this PR for the simple reason that it doesn't lead to the image file disappearing from the history for a sequence of commits, meaning that git will recognize it as a move.

That does imply that the WML part of the changes could be left for a separate commit, I suppose, but that would be kinda weird in itself… or perhaps what stevecotton said, making the changes to the scenario _first_ in a separate PR, and _then_ doing just the image move in this one (after rebasing it over the separate PR).

**Comment by sigurdfdragon** [comment]
@nemaara  I think you overestimate the gain to the game's initial impression made on new players that would be had by removing AOI. There are plenty of other places where gains can be made without resorting to the drastic and unprecedented removal of a finished campaign.
The Cliffs of Thoria branch of HttT was incomplete for a couple of years with a few threats of its removal hanging over its head before myself, @GregoryLundberg and othered finished it. I think AOI should at least get the same encouragement before any possibility of removal.

Also, I'm willing to work on this campaign in the 1.17 cycle.

As such, I'm absolutely opposed to this PR and the idea of removing AOI prior to 1.16

**Comment by nemaara** [comment]
I see that this is now beyond my ability to resolve with compromises. @Vultraz, since you're the project lead for now, could you make a decision on whether we should remove AOI or not?

**Comment by Pentarctagon** [comment]
Maybe I'm mis-remembering, but I don't think removal of a mainline campaign is entirely unprecedented?

**Comment by Vultraz** [comment]
Ok, short answer: yes, remove it.

Removing AOI won't magically make Wesnoth an amazing game or wow players, that is true. A lot more work needs to be done on *all* campaigns and a million other aspects of the game to truly bring it up to WOW level. And I don't recall AOI being a particularly bad campaign either, but it's just... bland. Tiny maps (I do like the designs, but damn they're tiny!), rather genetic and repetitive scenarios, and not a lot of lore or text.

In short, it's not *bad*, IMO, it's just... bland. Mainline campaigns are supposed to be of a very high quality. If we're going to raise the bar and attempt to make Wesnoth truly engaging, then "just ok" doesn't clear it. That applies here and to probably some other campaigns like THoT.

So, yeah, I say let's drop it and repurpose its assets.

**Comment by shikadiqueen** [comment]
Considering the campaign's complete lack of contribution to mainline's storyline or faction representation beyond spending too many scenarios hinting at something that's more properly handled by LoW, I'm in favour of AOI's removal.

And on the subject of improving it: anyone who feels strongly attached to Erlornas' trials and tribulations is free to do that with the campaign repurposed as an add-on. Raising mainline's content standards doesn't preclude trimming down excess material. It can even be reincorporated at a later time if it proves worth it. But right now it seems like that time and energy investment is better directed at stuff that has more substance to it.

> Maybe I'm mis-remembering, but I don't think removal of a mainline campaign is entirely unprecedented?

UtBS was removed once for less than 24 hours during the 1.5.x cycle simply because it was buggy and unmaintained.

**Comment by stevecotton** [comment]
Who will do what about the tutorial side of this? Removing AOI will leave a hole in the new-users area, and a new user is unlikely to look on the add-ons server for initial campaigns.

**Comment by CelticMinstrel** [comment]
They can still play AToTB and TSG as beginner campaigns (though I guess they're less tutorial-like) or the advanced tutorial that Yumi is proposing based on AoI.

**Comment by AI0867** [comment]
AToTB on easy is definitely beginner-material, but last time I checked TSG, the first part might be suitable, but the second part definitely was not.

Do we expect new players to move on to HttT or perhaps Liberty after that?

**Comment by GregoryLundberg** [comment]
I would suggest Liberty before HttT. There's a coupld points where a very new player might get bogged down in Liberty, but it shows off using terrain and how, sometimes, slow and carefull avoidance of conflict can be the best course of action. Seems better to me than sending everyone into the meat grinder of the Siege, 

I always felt Liberty should be before HttT but due to the timeline and due to difficulty. The only thing I'd suggest is to re-examine Liberty to ensure it does not presume knowledge (say, about who Ashevere is).

As to the hole, or lack of new-user campaigns ... are there any UMC which are short, SP campaigns, that fit the lore and could be adopted?

**Comment by sigurdfdragon** [comment]
@nemaara @Vultraz I can see and agree that my use of the word 'absolutely' in my post above was a mistake. It seems it has caused reflex style reactions and the overlooking of some of the points made.

Some of them have been answered. Some not.

@nemaara  Why the rush do this before 1.16? The impression I'm getting from you is that AOI is an intolerable eyesore that must immediately go.


>Maybe I'm mis-remembering, but I don't think removal of a mainline campaign is entirely unprecedented?

>UtBS was removed once for less than 24 hours during the 1.5.x cycle simply because it was buggy and unmaintained.

I'm talking about the unprecedentedness of it being removed with the full intention to ensure that it is not in the next stable release.

**Comment by shikadiqueen** [comment]
> I'm talking about the unprecedentedness of it being removed with the full intention to ensure that it is not in the next stable release.
The intention was to ensure that UtBS would not ship in 1.6.

**Comment by stevecotton** [comment]
Is there anything in the lore about merfolk-elf relations circa AOI?

According to TRoW, the merfolk have already been to the Great Continent circa 10 BW accompanying the First Crown Prince of Southbay, and that Southbay prince met some elves. It would seem odd if these were the only encounters between merfolk and elves.

In TRoW, the merfolk are trading with Southbay because they don't have underwater blacksmiths. It would also seem odd if the merfolk haven't been trading with the Great Continents' dwarves too.

Mainly to @sigurdfdragon: S5-S7 of AOI take place near Jotha, the merfolk town that's the first and last scenarios of Dead Water in 626-627 YW. As it's a good geographic location for the merfolk, it seems likely to me that Jotha would already exist before AOI, and that the mages of S5 would be at least on trading terms with Jotha - it also seems likely that the mages got there by ship, possibly even on one of the First Crown Prince of Southbay's voyages. The silver mages might make be working as cargo-carriers between Jotha and dwarves in the Heart Mountains, and S5 would give a reason for them to share information with the elves.

Rualsha's camp could be* at the mouth of the River Bork, with orcs arriving by ship from Tirigaz and by river-boat from the eastern orcish towns, and Rualsha needing to convince the other chiefs of his strength. If the other chiefs see the aftermath of S5, and the merfolk delay the ship from Tirigaz then the orcish horde falls apart, and S7 is the cleanup afterwards.

* The current bigmap.cfg puts Rualsha's camp further east, an inaccessible location that doesn't seem to have any logic, and which means S6's swamp is directly between S5 and S7.

**Comment by GregoryLundberg** [comment]
I've stepped away from following this for a while. But I'm again wondering why it's preferable to drop AOI rather than fix it.

**Comment by Pentarctagon** [comment]
The string freeze/1.16 release is delayed indefinitely, so that part isn't a concern.

I believe the larger point nemaara made in the linked thread though was that AOI is not really worth fixing in the first place.

**Comment by stevecotton** [comment]
The discussion about whether it was worth fixing seemed to end with nemaara saying that neither sigurdfdragon nor I had understood her core point. https://www.wesnoth.org/irclogs/2019/11/%23wesnoth-dev.2019-11-26.log from 18:11:23 until 18:21:21.

It's not clear whether I'm not seeing as far as her, or whether I'm seeing further than her; what I mean by that is whether
* I haven't seen what she's seen, or
* I've seen it, but I've also seen a reason to reject it, as in https://github.com/wesnoth/wesnoth/pull/4592#issuecomment-557947878

Please would someone who does understand nemaara's core point clarify it?

**Comment by Wedge009** [comment]
There's a very brief mention of Erlornas in the introduction to Legend of Wesmere:
https://github.com/wesnoth/wesnoth/blob/6c78d376fe8f4d6b62d6392bef738e500854851d/data/campaigns/Legend_of_Wesmere/scenarios/chapter1/01_The_Uprooting.cfg#L58

Given AOI is no longer official, is it worth removing that reference? Granted, it's not a significant reference but inevitably a relatively new player may wonder who Erlornas is, without knowing the history of the campaign he comes from.

**Comment by nemaara** [comment]
Nah, that's fine. Dropping references in mainline has been a thing for a while I think, it gives UMC things to work with as well. If we make mainline too closed off, that doesn't give content authors things to work off of.
